sky/clouds/service_catalog/common.py

```diff
@@ -111,8 +117,39 @@ def _get_modified_catalogs() -> List[str]:
     return modified_catalog_path_map
 
 
+class LazyDataFrame:
+    """A lazy data frame that reads the catalog on demand."""
```

Comment:
Add to docstr: We do it this way because ...  ? (IIUC, something about not every use of `import sky` needs to use the catalogs; maybe spell out those uses?)

Reply:
Added. Thanks!

Reply:
Added. Thanks!